### PR TITLE
Add a concatnatedDays method

### PR DESCRIPTION
**Pull request by TheBnl**
This method returns a similar output as the forWeekCombined method with the exception that it only combines days that are adjacent to each other.

Example output:
```
Array
(
    [monday] => Array
        (
            [opening_hours] => Spatie\OpeningHours\OpeningHoursForDay Object
                (
                    [openingHours:protected] => Array
                        (
                            [0] => Spatie\OpeningHours\TimeRange Object
                                (
                                    [start:protected] => Spatie\OpeningHours\Time Object
                                        (
                                            [hours:protected] => 0
                                            [minutes:protected] => 0
                                            [data:protected] => 
                                        )

                                    [end:protected] => Spatie\OpeningHours\Time Object
                                        (
                                            [hours:protected] => 0
                                            [minutes:protected] => 0
                                            [data:protected] => 
                                        )

                                    [data:protected] => 
                                )

                        )

                    [data:protected] => 
                )

            [days] => Array
                (
                    [0] => monday
                )

        )

    [tuesday] => Array
        (
            [opening_hours] => Spatie\OpeningHours\OpeningHoursForDay Object
                (
                    [openingHours:protected] => Array
                        (
                            [0] => Spatie\OpeningHours\TimeRange Object
                                (
                                    [start:protected] => Spatie\OpeningHours\Time Object
                                        (
                                            [hours:protected] => 9
                                            [minutes:protected] => 0
                                            [data:protected] => 
                                        )

                                    [end:protected] => Spatie\OpeningHours\Time Object
                                        (
                                            [hours:protected] => 18
                                            [minutes:protected] => 0
                                            [data:protected] => 
                                        )

                                    [data:protected] => 
                                )

                        )

                    [data:protected] => 
                )

            [days] => Array
                (
                    [0] => tuesday
                    [1] => wednesday
                    [2] => thursday
                )

        )

    [friday] => Array
        (
            [opening_hours] => Spatie\OpeningHours\OpeningHoursForDay Object
                (
                    [openingHours:protected] => Array
                        (
                            [0] => Spatie\OpeningHours\TimeRange Object
                                (
                                    [start:protected] => Spatie\OpeningHours\Time Object
                                        (
                                            [hours:protected] => 9
                                            [minutes:protected] => 0
                                            [data:protected] => 
                                        )

                                    [end:protected] => Spatie\OpeningHours\Time Object
                                        (
                                            [hours:protected] => 12
                                            [minutes:protected] => 0
                                            [data:protected] => 
                                        )

                                    [data:protected] => 
                                )

                        )

                    [data:protected] => 
                )

            [days] => Array
                (
                    [0] => friday
                )

        )

    [saturday] => Array
        (
            [opening_hours] => Spatie\OpeningHours\OpeningHoursForDay Object
                (
                    [openingHours:protected] => Array
                        (
                            [0] => Spatie\OpeningHours\TimeRange Object
                                (
                                    [start:protected] => Spatie\OpeningHours\Time Object
                                        (
                                            [hours:protected] => 9
                                            [minutes:protected] => 0
                                            [data:protected] => 
                                        )

                                    [end:protected] => Spatie\OpeningHours\Time Object
                                        (
                                            [hours:protected] => 18
                                            [minutes:protected] => 0
                                            [data:protected] => 
                                        )

                                    [data:protected] => 
                                )

                        )

                    [data:protected] => 
                )

            [days] => Array
                (
                    [0] => saturday
                )

        )

    [sunday] => Array
        (
            [opening_hours] => Spatie\OpeningHours\OpeningHoursForDay Object
                (
                    [openingHours:protected] => Array
                        (
                            [0] => Spatie\OpeningHours\TimeRange Object
                                (
                                    [start:protected] => Spatie\OpeningHours\Time Object
                                        (
                                            [hours:protected] => 0
                                            [minutes:protected] => 0
                                            [data:protected] => 
                                        )

                                    [end:protected] => Spatie\OpeningHours\Time Object
                                        (
                                            [hours:protected] => 0
                                            [minutes:protected] => 0
                                            [data:protected] => 
                                        )

                                    [data:protected] => 
                                )

                        )

                    [data:protected] => 
                )

            [days] => Array
                (
                    [0] => sunday
                )

        )

)
```

### CODE REVIEW

File: src/OpeningHours.php [hunk 1]
```diff
@@ -211,6 +211,26 @@ public function forWeekCombined(): array
...
     }
 
+    public function concatnatedDays(): array
+    {
+        $concatnatedDays = [];
+        $allOpeningHours = $this->openingHours;
+        foreach ($allOpeningHours as $day => $value) {
+            $previousDay = end($concatnatedDays);
+            if ($previousDay && (string) $previousDay['opening_hours'] === (string) $value) {
+                $key = array_key_last($concatnatedDays);
```
Comment (kylekatarnls) [inline src/OpeningHours.php]: `array_key_last` is not possible as `opening-hours` minimum compatible version is PHP 7.0.

File: src/OpeningHours.php [hunk 2]
```diff
@@ -211,6 +211,26 @@ public function forWeekCombined(): array
...
+                $concatnatedDays[$key]['days'][] = $day;
+            } else {
```
Comment (kylekatarnls) [inline src/OpeningHours.php]: Please prefer `continue;` than else.

File: src/OpeningHours.php (hunk 1 above)
Comment (TheBnl) [reply src/OpeningHours.php]: As i'm using end() i can use the key() method as well 👍

**Comment by TheBnl** [comment]
Still need to add some tests

**Comment by TheBnl** [comment]
Test is added! 👍

**Comment by kylekatarnls** [comment]
This is very good. Thanks a lot.

concatnated => concatenated and concatenatedDays is not very consistent with other methods names. I will try to find something.

The last thing that tickle me is tests for concatnatedDays and forWeekCombined could be swapped without consequence. We should have an assertion in both with non-consecutive common ranges to highlight the difference between both.

**Comment by TheBnl** [comment]
Awesome! Thanks for merging 👍

**Pull request merged**